apl-db-updater/src/main/resources/db/migration/mariadb/apl/V1_0__Initialize_tables.sql

```diff
@@ -0,0 +1,1295 @@
+  `height` int(11) NOT NULL,
+  `generation_signature` binary(32) NOT NULL,
+  `block_signature` binary(64) NOT NULL,
+  `payload_hash` binary(32) NOT NULL,
+  `generator_id` bigint(20) NOT NULL,
+  `timeout` int(11) NOT NULL DEFAULT 0,
+  UNIQUE KEY `block_id_idx` (`id`),
+  UNIQUE KEY `block_height_idx` (`height`),
+  UNIQUE KEY `block_timestamp_idx` (`TIMESTAMP`),
+  KEY `block_generator_id_idx` (`generator_id`),
+  CONSTRAINT `chk_timeout` CHECK (`timeout` >= 0)
+) ;
+
+
+
+CREATE TABLE IF NOT EXISTS `block_index`
+(
+    `block_id`     bigint(20) NOT NULL,
+    `block_height` int(11)    NOT NULL,
+    UNIQUE KEY `block_index_block_id_idx` (`block_id`),
+    UNIQUE KEY `block_index_block_height_idx` (`block_height`)
+) ;
+
+
+
+CREATE TABLE IF NOT EXISTS `buy_offer`
+(
+    `db_id`             bigint(20) unsigned NOT NULL AUTO_INCREMENT primary key,
+    `id`                bigint(20)  NOT NULL,
+    `currency_id`       bigint(20)  NOT NULL,
+    `account_id`        bigint(20)  NOT NULL,
+    `rate`              bigint(20)  NOT NULL,
+    `unit_limit`        bigint(20)  NOT NULL,
+    `supply`            bigint(20)  NOT NULL,
+    `expiration_height` int(11)     NOT NULL,
+    `creation_height`   int(11)     NOT NULL,
+    `transaction_index` smallint(6) NOT NULL,
+  `transaction_height` int(11) NOT NULL,
+  `height` int(11) NOT NULL,
+  `latest` tinyint(1) NOT NULL DEFAULT 1,
+  `deleted` tinyint(1) NOT NULL DEFAULT 0,
+  UNIQUE KEY `buy_offer_id_idx` (`id`,`height`),
+  KEY `buy_offer_currency_id_account_id_idx` (`currency_id`,`account_id`,`height`),
+  KEY `buy_offer_rate_height_idx` (`rate`,`creation_height`),
+  KEY `buy_offer_height_id_idx` (`height`,`id`)
+) ;
+
+
+
+CREATE TABLE IF NOT EXISTS `currency`
+(
+    `db_id`                  bigint(20) unsigned NOT NULL AUTO_INCREMENT primary key,
+    `id`                     bigint(20) NOT NULL,
+    `account_id`             bigint(20) NOT NULL,
+    `name`                   varchar(191)  NOT NULL,
+    `name_lower`             varchar(191)  NOT NULL,
+    `code`                   varchar(191)  NOT NULL,
+    `description`            varchar(1000)  DEFAULT NULL,
+    `type`                   int(11)    NOT NULL,
+    `initial_supply`         bigint(20) NOT NULL DEFAULT 0,
+    `reserve_supply`         bigint(20) NOT NULL,
+  `max_supply` bigint(20) NOT NULL,
+  `creation_height` int(11) NOT NULL,
+  `issuance_height` int(11) NOT NULL,
+  `min_reserve_per_unit_atm` bigint(20) NOT NULL,
+  `min_difficulty` tinyint(4) NOT NULL,
+  `max_difficulty` tinyint(4) NOT NULL,
+  `ruleset` tinyint(4) NOT NULL,
+  `algorithm` tinyint(4) NOT NULL,
+  `decimals` tinyint(4) NOT NULL DEFAULT 0,
+  `height` int(11) NOT NULL,
+  `latest` tinyint(1) NOT NULL DEFAULT 1,
+  `deleted` tinyint(1) NOT NULL DEFAULT 0,
+  UNIQUE KEY `currency_id_height_idx` (`id`,`height`),
+  KEY `currency_account_id_idx` (`account_id`),
+  KEY `currency_name_idx` (`name_lower`,`height`),
```

Comment:
currency_name_idx, currency_code_idx  will not work properly with default collation.